src/envoy/http/mixer/filter.cc

```diff
@@ -122,6 +121,9 @@ FilterHeadersStatus Filter::decodeHeaders(HeaderMap& headers, bool) {
       [this](const Status& status) { completeCheck(status); });
   initiating_call_ = false;
 
+  // Remove Istio authentication header after Check()
+  Envoy::Utils::Authentication::ClearResultInHeader(&headers);
```

Comment:
The right place to remove should be in 
completeCheck()

Reply:
Can we have test for this removing?

Reply:
It has been placed inside completeCheck(). As there is no testing for the mixer filter, per the discussion, this code in the mixer filter is not tested.